src/app/clusters/operational-state-server/operational-state-server.h

```diff
@@ -269,6 +269,8 @@ class Delegate
     /**
      * Fills in the provided MutableCharSpan with the phase at index `index` if there is one,
      * or returns CHIP_ERROR_NOT_FOUND if the index is out of range for the list of phases.
+     * If fills in the provided MutableCharSpan with the phase at index `0` and returns CHIP_ERROR_NOT_FOUND,
+     * it represents PhaseList attribute is an empty list, the SDK will set PhaseList attribute value to null.
```

Comment:
How about:
```suggestion
     *
     * If CHIP_ERROR_NOT_FOUND is returned for index 0, that indicates that the PhaseList attribute is null
     * (there are no phases defined at all).
     *
```
and the same in all the other places where this got copy/pasted.

Reply:
This is probably more of a spec question than an SDK question, but why have a nullable list to begin with? Per 17.18.1 

```
Composite data types that have a length (i.e. octet string and list), and derived types that have those as the base type, SHALL NOT differentiate semantically between the null value and the empty (zero length) value. 
```

This seems like extra wrappers for no good reason. Or is this just for backwards compatibility?

Reply:
The spec could have gone either way here....  There are various opinions about which is better (e.g. null is one less byte on the wire than an empty list is).  But for purposes of the SDK, this is what the spec says, and yes, this has already shipped as nullable, so at this point we can't really change the spec.